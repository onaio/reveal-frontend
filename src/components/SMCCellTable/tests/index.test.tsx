import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import SMCTableCell from '..';

describe('src/components/SMCTableCell', () => {
  it('renders correctly', () => {
    const cell = {
      row: {
        original: { plan_id: 'plan_id', jurisdiction_id: 'jurisdiction_id' },
      },
    };

    const props: any = {
      cell,
      cellValue: 'CellValue',
      hasChildren: true,
    };
    const wrapper = shallow(<SMCTableCell {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot('has children');

    wrapper.setProps({ hasChildren: false });
    wrapper.update();
    expect(toJson(wrapper)).toMatchSnapshot('has not children');
  });

  it('renders correctly when is_virtual_jurisdiction and has not children', () => {
    const cell = {
      row: {
        original: {
          is_virtual_jurisdiction: true,
          jurisdiction_id: 'jurisdiction_id',
          plan_id: 'plan_id',
        },
      },
    };

    const props: any = {
      cell,
      cellValue: 'CellValue',
      hasChildren: false,
    };
    const wrapper = shallow(<SMCTableCell {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot('has children');
  });
});
