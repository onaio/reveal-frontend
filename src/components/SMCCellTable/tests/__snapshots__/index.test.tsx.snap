// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/SMCTableCell renders correctly when is_virtual_jurisdiction and has not children: has children 1`] = `
<div>
  <span
    className="plan-jurisdiction-name name-label"
  >
    CellValue
  </span>
</div>
`;

exports[`src/components/SMCTableCell renders correctly: has children 1`] = `
<div>
  <Link
    to=""
  >
    CellValue
  </Link>
</div>
`;

exports[`src/components/SMCTableCell renders correctly: has not children 1`] = `
<div>
  <span
    className="plan-jurisdiction-name main-span"
  >
    <span
      className="plan-jurisdiction-name btn-link"
    >
      <Link
        to="undefined/jurisdiction_id/map"
      >
        <FontAwesomeIcon
          icon={
            Array [
              "fas",
              "map",
            ]
          }
        />
      </Link>
    </span>
      
    <Link
      to="undefined/jurisdiction_id/map"
    >
      CellValue
    </Link>
  </span>
</div>
`;
